src/test/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessorTest.java

```diff
@@ -25,36 +33,51 @@
     @Mock private IMessageReceiver messageReceiver;
     @Mock private SendLetterService sendLetterService;
     @Mock private IMessage message;
+    @Mock private AppInsights insights;
 
     @Before
-    public void setUp() throws Exception {
-        this.processor = new MessageProcessor(() -> messageReceiver, sendLetterService);
+    public void setUp() {
+        processor = new MessageProcessor(() -> messageReceiver, sendLetterService);
+
+        ReflectionTestUtils.setField(processor, "insights", insights);
     }
 
     @Test
     public void should_complete_message_if_passed_function_returns_success() throws Exception {
         // given
         given(sendLetterService.send(any())).willReturn(SUCCESS);
         given(messageReceiver.receive()).willReturn(message);
+        given(message.getEnqueuedTimeUtc()).willReturn(Instant.now().minusSeconds(100));
 
         // when
         processor.process();
 
         // then
-        verify(messageReceiver, times(1)).complete(any());
+        verify(messageReceiver).complete(any());
+        verify(insights).trackMessageReceivedFromServiceBus(any(Duration.class), eq(true));
+        verify(insights).trackMessageReceived(anyString(), anyLong());
+        verify(insights).markMessageHandled(anyString(), anyLong());
+        verify(insights).trackMessageCompletedInServiceBus(any(Duration.class), eq(true));
+        verifyNoMoreInteractions(insights);
```

Comment:
I think it would be nice to have a dedicated suite of tests dedicated to insights

Reply:
Will be duplicating work since insights are knotted within each component and depends on outcome - tracks relevant action.

The dead-simple test suite for app insights is pending - can add as a separate PR to reduce the volume